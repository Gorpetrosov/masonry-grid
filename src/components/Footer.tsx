import config from "../config";

const currentYear = new Date().getFullYear();

const Footer = () => {
    return (
        <footer className="layout-footer">
            <p>Â© {currentYear} {config.APP_NAME}</p>
        </footer>
    )
}

export default Footer;