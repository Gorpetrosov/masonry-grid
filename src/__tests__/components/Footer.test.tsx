import { render, screen } from '@testing-library/react';
import { describe, test, expect, vi, beforeEach } from 'vitest';
import Footer from "../../components/Footer.tsx";

vi.mock('../config', () => ({
    default: { APP_NAME: 'Test App' }
}));

describe('Footer Component', () => {
    beforeEach(() => {
        vi.useFakeTimers();
        vi.setSystemTime(new Date(2023, 0, 1)); // Jan 1, 2023
    });

    afterEach(() => {
        vi.useRealTimers();
        vi.resetAllMocks()
    });

    test('memoization prevents unnecessary renders', async () => {
        const { rerender } = render(<Footer />);
        const initialHTML = screen.getByRole('contentinfo').innerHTML;

        rerender(<Footer />);

        expect(screen.getByRole('contentinfo').innerHTML).toBe(initialHTML);
    });
});