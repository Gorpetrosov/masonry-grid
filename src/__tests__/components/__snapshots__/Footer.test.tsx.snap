// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > matches snapshot 1`] = `
<div>
  <footer
    class="layout-footer"
  >
    <p>
      Â© 
      2025
       
      Masonry APP
    </p>
  </footer>
</div>
`;
